rohmu/object_storage/azure.py

```diff
@@ -113,6 +113,42 @@ def __init__(
         self.container = self.get_or_create_container(self.container_name)
         self.log.debug("AzureTransfer initialized, %r", self.container_name)
 
+    @staticmethod
+    def conn_string(
+        account_name: str,
+        account_key: Optional[str],
+        azure_cloud: Optional[str],
+        host: Optional[str],
+        port: Optional[int],
+        is_secure: bool,
+    ) -> str:
+        if not host and not port:
+            try:
+                endpoint_suffix = ENDPOINT_SUFFIXES[azure_cloud]
+            except KeyError:
+                raise InvalidConfigurationError(f"Unknown azure cloud {repr(azure_cloud)}")
+
+            conn_str = (
+                "DefaultEndpointsProtocol=https;"
+                f"AccountName={account_name};"
+                f"AccountKey={account_key};"
+                f"EndpointSuffix={endpoint_suffix}"
+            )
+        else:
+            print(host)
+            print(port)
+            if not host or not port:
+                raise InvalidConfigurationError("Custom host and port must be specified together")
+
+            protocol = "https" if is_secure else "http"
+            conn_str = (
+                f"DefaultEndpointsProtocol={protocol};"
+                f"AccountName={account_name};"
+                f"AccountKey={account_key};"
+                f"BlobEndpoint={protocol}://{host}:{port}/{account_name};"
+            )
+        return conn_str
```

Comment:
There is duplication of 3/4 lines of `conn_str`. I think this can be rewritten to be readable and without duplication the same time.